### ISSUE TITLE
Contact form doesn't work and gives me an error

### ISSUE THREAD

**Issue by ghost** [jadwigo/SimpleForms#18]
I have installed version 1.0 of SimpleForms with Bolt 2.1.5. I can edit the config file and make the form visible on the contact page. But the form gives an error when I want to send email (see screenshot). I haven't edit the default config file of Bolt (mailoptions). On an other Bolt install I also use SimpleForms and that works fine without mailoptions. What am I doing wrong?

![screenshot](https://cloud.githubusercontent.com/assets/9332608/7104877/a77993f4-e0fd-11e4-8975-6cb9a28d19e2.jpg)

**Comment by GwendolenLynch** [comment]
Probably differing Bolt versions.

Have a look at [these config options](https://github.com/bolt/bolt/blob/master/app/config/config.yml.dist#L248-L260) and set one on.

You can probably just get away with:

``` yaml
mailoptions:
    transport: smtp
    spool: true
    host: localhost
```

**Comment by ghost** [comment]
This doesn't work. Even with all mail settings (see below), I get the same error.

mailoptions:
   transport: smtp
   spool: true
   host: send.one.com
   username: username
   password: password
   port: 465
   auth_mode: plain
   encryption: ssl

**Comment by GwendolenLynch** [comment]
I would guess that the shared host is blocking out going ports… or something similar. 

In Bolt 2.1 there is a super secret test for email configuration, it will still probably throw the same error but will be quicker to test…

You can hit `http://myboltsite/async/email/test/me` (you **must** be logged in) and that will attempt to send an email as the Bolt site to the configured email address of the logged in user.

**Comment by ghost** [comment]
This also gives an error:

Done{"error":{"type":"Swift_TransportException","message":"Connection could not be established with host send.one.com [Verbinding is verlopen #110]","file":"\/customers\/9\/a\/b\/coenders-montage.nl\/httpd.www\/vendor\/swiftmailer\/swiftmailer\/lib\/classes\/Swift\/Transport\/StreamBuffer.php","line":265}}

**Comment by GwendolenLynch** [comment]
Yes, hence the "will still probably throw the same error but will be quicker to test" bit :grin: 

You hosting set up is the problem, `send.one.com` is not reachable from your web server and I'll put money on it being your hosts firewall.

If that can be resolved, rather than filling in a form every time you can just F5 that URL in your browser to test if your host has fixed the problem. That part is just a "nice to have".

**Comment by ghost** [comment]
I contacted the provider. It turns out that they use the mentioned settings for webmail. They have separate settings (different port and url, no authentication and encryption) for mail scripts. It now works like it should!

**Comment by GwendolenLynch** [comment]
Yay! Good stuff… "working" is always a good result :grin:

**Comment by ghost** [comment]
It's nice that the form is working now, but there is one thing left...

Received messages have the form owner address as 'from address' instead of the address of the actual sender. In the config file I had to fill in the recipient_email and the form takes over that address to display as from address. That is confusing. How can I set the sender address for this?

**Comment by GwendolenLynch** [comment]
Is what you're looking for the `from_email:` and  `from_name:` config keys?

**Comment by ghost** [comment]
I have seen those keys, but what to fill in? When someone uses the form, they use their own email address. How can I use that to display as from address when the message is received? I can use this in the body of the message, but in the header the from address is the same as the 'to address'.

**Comment by GwendolenLynch** [comment]
Honestly, I am not sure. I don't use SimpleForms any more.  I wrote [BoltForms](https://github.com/GawainLynch/bolt-extension-boltforms) last year to overcome some of these needs.  It's a bit more complex in the configuration, but gives you full access to Symfony Forms configuration, the latest reCAPTCHA API, ability to write out Contenttype records and a whole lot more. 

The plan is to remove most of the code in SimpleForms and have it use the API provided by BoltForms, but @jadwigo and I just haven't had time to do it.

**Comment by ghost** [comment]
I think I'll use BoltForms then. Just installed it. Looks familiar with SimpleForms. Does this extension use smtp settings from the config file? I have to use from_name and from_email (keys that use data from the sender). Is there a file to markup the message that is received? I had this with SimpleForms:

<b>Naam</b><br />
{{form.name}}<br /><br />
<b>E-mail</b><br />
{{ form.email }}<br /><br />
<b>Bericht</b><br />
{{form.message}}

**Comment by GwendolenLynch** [comment]
Sure is, just haven't documented it properly/yet :grin: 

You can see the [default templates](https://github.com/GawainLynch/bolt-extension-boltforms/tree/master/assets) for fields, etc.

If you want to change the path or name of any of them, you can add this to the config:

``` yaml
templates:
    form: boltforms_form.twig
    email: boltforms_email.twig
    subject: boltforms_email_subject.twig
```

**Comment by ghost** [comment]
Thanks! Everything works fine now! I will uninstall SimpleForms.

Where can I edit the actual form? It looks like this (screenshot):

![screenshot](https://cloud.githubusercontent.com/assets/9332608/7113468/706120c6-e1d4-11e4-8218-486aebc30f14.jpg)

I just need some breaks to display the fields below their titles.

**Comment by GwendolenLynch** [comment]
Sorry it took so long to get back to you, helping someone with git fun  :grin: 

I have tagged a new version of BoltForms and in there, I have split up the row generation and it now does:

``` twig
                {{ form_label(form[key]) }}
                {{ form_errors(form[key]) }}
                {{ form_widget(form[key]) }}
```

So you can copy in the new Twig template changes into yours and style as you like.

**Comment by ghost** [comment]
Thanks! The only thing that doesn't work is the feedback after sending a message successfully. The feedback doesn't appear...

**Comment by GwendolenLynch** [comment]
Derp! That was an untested PR that I merged.  Fixed now in v2.1.4 which should appear in the next 30 minutes, but if you're using a custom template, see the [changes in this commit](https://github.com/GawainLynch/bolt-extension-boltforms/commit/12a4f03b0c36f156e98849295d16d76f04e3b480)

**Comment by jadwigo** [comment]
I guess this issue is closed now?
